undercrawler/dupe_filter.py

```diff
@@ -0,0 +1,23 @@
+import re
+from copy import deepcopy
+
+from scrapy.utils.url import canonicalize_url
+from scrapy_splash import SplashAwareDupeFilter
+
+
+class DupeFilter(SplashAwareDupeFilter):
+    """
+    Consider same urls with and without www as duplicates.
+    """
+    def request_fingerprint(self, request):
+        if not request.meta.get('_splash_processed'):
+            # canonicalize_url required only due to a bug (?) in scrapy_splash
```

Comment:
This should be fixed in https://github.com/scrapy-plugins/scrapy-splash/commit/8ca86351b0a6a40b05b58f096e09650de142beed.